### PR TITLE
Optimize interval_list file gen

### LINKED ISSUE #28: exome-seq: Optimize BedToIntervalList steps
There are 2 steps in exomeseq-01-preprocessing that produce interval lists from bed files:

https://github.com/Duke-GCB/bespin-cwl/blob/341db4b3c65134f4911a9bfae9ff856a69d77f86/workflows/exomeseq-01-preprocessing.cwl#L152-L177

exomeseq-01-preprocessing is scattered, so this step would be repeated on the same data for every sample. Assuming it takes some measurable amount of time, this should be optimized to either precompute the interval lists and store them with the rest of the precomputed data (like genome indices) or pull the step out so that it does not scatter.

### LINKED ISSUE #29: exome-seq: Correct output type for hs_metrics_dir
hs_metrics_dir type should be `Directory` not `File`.
https://github.com/Duke-GCB/bespin-cwl/blob/a828902416018c2523903d9ffa78ae0a14d8a3ee/workflows/exomeseq.cwl#L79-L81

### PR DESCRIPTION
**Pull request by johnbradley**
Moves generating interval lists out of the scatter step in exomeseq.cwl.
Fixes #28

### CODE REVIEW

File: workflows/exomeseq.cwl [hunk 1]
```diff
@@ -96,12 +96,40 @@ outputs:
...
+      - class: ResourceRequirement
+        coresMin: 1
+        ramMin: 4000
+        outdirMin: 12000
+        tmpdirMin: 12000
+    in:
+      input_file: primary_intervals
+      reference_sequence: reference_genome
+    out:
+      - output_interval_list_file
```
Comment (dleehr) [inline workflows/exomeseq.cwl]: Can these tool steps be abstracted into a workflow like `exomeseq-00-prepare-reference-data` for consistency with the rest of the steps?

**Comment by dleehr** [comment]
Sorry this got stale, but it looks like it needs some conflicts resolved before we can merge

**Comment by dleehr** [comment]
@johnbradley Just a reminder that this PR is outstanding with merge conflicts.

**Comment by johnbradley** [comment]
@dleehr This should be good for review/merge now.

**Pull request merged**